app/types/app_version_type.rb

```diff
@@ -0,0 +1,15 @@
+class AppVersionType < ActiveRecord::Type::Value
+  def cast(value)
+    value.to_s.present? ? AppVersion.new(value) : nil
```

Comment:
This is a little nuanced and maybe overkill, but I want to try to parse it as long as the value isn't `nil` - i.e. `false` should parse and rescue. It just feels like we shouldn't incur the rescue overhead for a "this is clearly nil and not worth attempting to parse" case.

Speaking of which maybe this isn't the most revealing impl of that:

```suggestion
    AppVersion.new(value) unless value.nil?
```

This differs from serialize and deserialize because it's handling user input, and any value is possible so we want to be pretty liberal in what we accept, but it just feels good not to fall through a raise/rescue codepath unless the user screws up.